lib/apollo-federation/entity.rb

```diff
@@ -9,5 +9,25 @@ class Entity < GraphQL::Schema::Union
     def self.resolve_type(object, context)
       context[object]
     end
+
+    # The main issue here is the fact that an union in GraphQL can't be an interface
+    # according to the [spec](https://spec.graphql.org/October2021/#sec-Unions.Type-Validation),
+    # but at the same time, according to the Federation spec, an interface can be an Entity,
+    # and an Entity is an union. Therefore, we have to extend this validation to allow interfaces
+    # (more precisely Modules) as possible types.
+    def self.assert_valid_union_member(type_defn)
```

Comment:
I think it would be good if we can monkey patch as little as possible. 
Also, double check that the module is indeed a GraphQL::Interface? 
Suggestion

```
def self.assert_valid_union_member(type_defn)
      if type_defn.is_a?(Module) &&
          type_defn.included_modules.include?(GraphQL::Schema::Interface)
        # This is the only change we made to the original method
        # to allow interfaces as possible types
      else
        super(type_defn)
      end
    end

```

Reply:
Thanks for the suggestion. I've actually changed it a bit to check straight for `ApolloFederation::Interface`, which makes most sense in this context.